src/math/warpExposure.cc

```diff
@@ -350,13 +352,326 @@ namespace {
         return std::abs(dSrcA.getX()*dSrcB.getY() - dSrcA.getY()*dSrcB.getX());
     }
 
+
+    template<typename DestImageT, typename SrcImageT>
+    int doWarpCovariance(
+      DestImageT &destImage,                      ///< remapped %image
+      SrcImageT const &srcImage,                  ///< source %image
+      afwMath::detail::PositionFunctor const &computeSrcPos,   ///< Functor to compute source position
+          ///< called with dest row, column; returns source position (as a Point2D)
+      afwMath::WarpingControl const &control,     ///< warping parameters
+      afwImage::Image<double> &covImage,           ///< covariance matrix
+      typename DestImageT::SinglePixel padValue   ///< value to use for undefined pixels
+    ) {
+      if (afwMath::details::isSameObject(destImage, srcImage)) {
+          throw LSST_EXCEPT(pexExcept::InvalidParameterError,
+              "destImage is srcImage; cannot warp in place");
+      }
+      if (destImage.getBBox(afwImage::LOCAL).isEmpty()) {
+          return 0;
+      }
+      // if src image is too small then don't try to warp
+      try {
+          PTR(afwMath::SeparableKernel) warpingKernelPtr = control.getWarpingKernel();
+          warpingKernelPtr->shrinkBBox(srcImage.getBBox(afwImage::LOCAL));
+      } catch(...) {
+          for (int y = 0, height = destImage.getHeight(); y < height; ++y) {
+              for (typename DestImageT::x_iterator destPtr = destImage.row_begin(y), end = destImage.row_end(y);
+                  destPtr != end; ++destPtr) {
+                  *destPtr = padValue;
+              }
+          }
+          return 0;
+      }
+      PTR(afwMath::SeparableKernel) warpingKernelPtr = control.getWarpingKernel();
+
+      std::shared_ptr<afwMath::LanczosWarpingKernel const> const lanczosKernelPtr = std::dynamic_pointer_cast<afwMath::LanczosWarpingKernel>(warpingKernelPtr);
+
+      int numGoodPixels = 0;
+
+      // Get the source MaskedImage and a pixel accessor to it.
+      int const srcWidth = srcImage.getWidth();
+      int const srcHeight = srcImage.getHeight();
+      int const destWidth = destImage.getWidth();
+      int const destHeight = destImage.getHeight();
+
+      std::vector<afwGeom::Point2D> _srcPosList(1 + destWidth);
+
+      afwMath::detail::WarpAtOnePoint<DestImageT, SrcImageT> warpAtOnePoint(srcImage, control, padValue);
+
+      std::vector<afwGeom::Point2D>::iterator srcPosView = _srcPosList.begin() + 1;
+      double kernelFactor = 0.0;
+      for (int col = -1; col < destWidth; ++col) {
+          srcPosView[col] = computeSrcPos(col, -1);
+      }
```

Comment:
This bit of code is copied over from the original. The original code had the comment
 // A cache of pixel positions on the source corresponding to the previous or current row
 // of the destination image.
 // The first value is for column -1 because the previous source position is used to compute relative area
 // To simplify the indexing, use an iterator that starts at begin+1, thus:
 // srcPosView = _srcPosList.begin() + 1
 // srcPosView[col-1] and lower indices are for this row
 // srcPosView[col] and higher indices are for the previous row